pkg/component/component.go

```diff
@@ -372,17 +421,36 @@ func PushLocal(client *occlient.Client, componentName string, applicationName st
 	}
 	targetPath := fmt.Sprintf("%s/src", s2iSrcPath)
 
+	// If there are files identified as deleted, propogate them to the component pod
+	if len(delFiles) > 0 {
+		s := log.Spinner("deleting files from component")
+		defer s.End(false)
+		glog.V(4).Infof("propogating deletion of files %s to pod", strings.Join(delFiles, " "))
+		err := client.PropogateDeletes(pod.Name, delFiles, getS2IPaths(pod.Spec.Containers[0].Env))
```

Comment:
It would be good to have explained somewhere why we are deleting files from multiple directories (s2ipaths)